03week/ticTacToe.js

```diff
@@ -24,23 +24,75 @@ function printBoard() {
 }
 
 function horizontalWin() {
-  // Your code here
-}
+  if((board[0][0]==='X' && board[0][1]==='X' && board[0][2]==='X') 
+  || (board[1][0]==='X' && board[1][1]==='X' && board[1][2]==='X')
+  || (board[2][0]==='X' && board[2][1]==='X' && board[2][2]==='X')){
+    return true;
+  }
+  
+  if((board[0][0]==='O' && board[0][1]=='O' && board[2][0]==='O')
+  || (board[1][0]==='O' && board[1][0]==='O' && board[1][2]==='O')
+  ||(board[2][0]==='O' && board[2][1]==='O' && board[2][2]==='O')){
+    return true;
+  }
+    }
 
 function verticalWin() {
-  // Your code here
+   if((board[0][0]==='X' &&board[1][0]==='X' && board[2][0]==='X')
+   ||(board[0][1]==='X' && board[1][1]==='X' && board[2][1]==='X')
+   ||(board[0][2]==='X' && board[1][2]==='X' && board[2][2]==='X'))
+   return true;
+   if((board[0][0]==='O' && board[1][0]==='O' && board[2][0]==='O')
+   || (board[0][1]==='O' && board[1][1]==='O' && board[1][2]==='O')
+   ||(board[0][2]==='O' && board[1][2]==='O' && board[2][2]==='O')){
+    return true;
+  }
 }
 
 function diagonalWin() {
-  // Your code here
+  if((board[0][0]==='X' && board[1][1]==='X' && board[2][2]==='X')
+  || (board[2][0]==='X' && board[2][1]==='X' && board[2][2]==='X')){
+    return true;
+
+  }
+  if((board[0][0]==='O' && board[1],[1]==='O' && board[2],[2]==='O')
+  || (board[2][0]==='O' && board[2],[1]==='O' && board[2],[2]==='O')){
+    return true;
+
+  }
+  
 }
 
+
+
 function checkForWin() {
-  // Your code here
+  
+  if(horizontalWin()===true){
+    return true
+  }
+  if(verticalWin()===true){
+    return true
+  }
+  if(diagonalWin()===true){
+    return true
+  }
+}
+
+function switchPlayer(){
+if(playerTurn==='X'){
+  playerTurn='O'
+}
+else{
+  playerTurn='X'
+}
 }
 
 function ticTacToe(row, column) {
-  // Your code here
+  board[row][column]=playerTurn;
+  horizontalWin();
+  verticalWin();
+  diagonalWin();
+  switchPlayer();
 }
```

Comment:
you need to also check for valid input. User input for row and column needs to be between 0-2. check if the selected row/column combination is occupied.